File: tests/fixtures/java/BranchIfTest.java
class BranchIfTest
{
    public static int ifAcmpEq(String value1, String value2)
    {
	return value1 != value2 ? 0 : 1;
    }

    public static int ifAcmpNe(String value1, String value2)
    {
	return value1 == value2 ? 0 : 1;
    }

    public static int ifIcmpEq(int value1, int value2)
    {
	return value1 != value2 ? 0 : 1;
    }

    public static int ifIcmpNe(int value1, int value2)
    {
	return value1 == value2 ? 0 : 1;
    }

    public static int ifIcmpLt(int value1, int value2)
    {
	return value1 >= value2 ? 0 : 1;
    }

    public static int ifIcmpGe(int value1, int value2)
    {
	return value1 < value2 ? 0 : 1;
    }

    public static int ifIcmpGt(int value1, int value2)
    {
	return value1 <= value2 ? 0 : 1;
    }

    public static int ifIcmpLe(int value1, int value2)
    {
	return value1 > value2 ? 0 : 1;
    }

    public static int ifLcmpGraterThan(long value1, long value2)
    {
        return value1 > value2 ? 0 : 1;
    }

    public static int ifLcmpLessThan(long value1, long value2)
    {
        return value1 < value2 ? 0 : 1;
    }

    public static int ifLcmpEqualsTo(long value1, long value2)
    {
        return value1 == value2 ? 0 : 1;
    }
}

File: tests/BranchIfTest.php
<?php
namespace PHPJava\Tests;

use PHPJava\Kernel\Types\_Long;

class BranchIfTest extends Base
{
    protected $fixtures = [
        'BranchIfTest',
    ];

    private function call($method, $value1, $value2)
    {
        $calculatedValue = $this->initiatedJavaClasses['BranchIfTest']
            ->getInvoker()
            ->getStatic()
            ->getMethods()
            ->call($method, $value1, $value2);

        return $calculatedValue->getValue();
    }

    public function testIfAcmpEq()
    {
        $actual = $this->call('ifAcmpEq', 'value1', 'value2');
        $this->assertEquals(0, $actual);
    }

    public function testIfAcmpNe()
    {
        $actual = $this->call('ifAcmpNe', 'value1', 'value2');
        $this->assertEquals(1, $actual);
    }

    public function testIfIcmpEq()
    {
        $actual = $this->call('ifIcmpEq', 5, 3);
        $this->assertEquals(0, $actual);
    }

    public function testIfIcmpNe()
    {
        $actual = $this->call('ifIcmpNe', 5, 3);
        $this->assertEquals(1, $actual);
    }

    public function testIfIcmpLt()
    {
        $actual = $this->call('ifIcmpLt', 5, 3);
        $this->assertEquals(0, $actual);
    }

    public function testIfIcmpGe()
    {
        $actual = $this->call('ifIcmpGe', 5, 3);
        $this->assertEquals(1, $actual);
    }

    public function testIfIcmpGt()
    {
        $actual = $this->call('ifIcmpGt', 5, 3);
        $this->assertEquals(1, $actual);
    }

    public function testIfIcmpLe()
    {
        $actual = $this->call('ifIcmpLe', 5, 3);
        $this->assertEquals(0, $actual);
    }

    public function testIfLcmpGraterThan()
    {
        $actual = $this->call('ifLcmpGraterThan', _Long::get(5), _Long::get(3));
        $this->assertEquals(0, $actual);
    }

    public function testIfLcmpLessThan()
    {
        $actual = $this->call('ifLcmpLessThan', _Long::get(5), _Long::get(3));
        $this->assertEquals(1, $actual);
    }

    public function testIfLcmpNotEqualsTo()
    {
        $actual = $this->call('ifLcmpEqualsTo', _Long::get(5), _Long::get(3));
        $this->assertEquals(1, $actual);
    }

    public function testIfLcmpEqualsTo()
    {
        $actual = $this->call('ifLcmpEqualsTo', _Long::get(5), _Long::get(5));
        $this->assertEquals(0, $actual);
    }
}
